src/math/Vector3.js

```diff
@@ -312,27 +312,19 @@ Object.assign( Vector3.prototype, {
 
 	},
 
-	project: function () {
+	project: function ( camera ) {
 
-		var matrix = new Matrix4();
-
-		return function project( camera ) {
+		return this.applyMatrix4( camera.matrixWorldInverse ).applyMatrix4( camera.projectionMatrix );
```

Comment:
Are you sure using `camera.matrixWorldInverse` is safe? I wonder if there are scenarios where this matrix might not be up to date (or in sync with `camera.matrixWorld`)...

Reply:
yes, i am sure. camera.matrixWorldInverse was updated by the camera itself automatically in Camera.updateMatrixWorld. and it should be sync with camera.matrixWorld as the name implies.

Reply:
Yep! https://github.com/mrdoob/three.js/blob/dev/src/cameras/Camera.js#L60